### PR TITLE
use netnsid for detecting cgroup networks; 

### LINKED ISSUE #5383: Docker container networking
We have a bunch of docker containers with names.  Is it possible to show the name instead of the container id?  Secondly, each of these containers have an veth interface and it would be nice to match those up to the container name - is that possible?

### PR DESCRIPTION
**Pull request by ktsaou**
##### Summary

fixes #5383

This PR allows `cgroup-network` to also use `ip link` output to detect container network interfaces. `cgroup-network` now uses `netnsid` to assign network interfaces to containers.

##### Component Name

collectors/cgroups

### CODE REVIEW

File: collectors/cgroups.plugin/cgroup-network-helper.sh [hunk 1]
```diff
@@ -184,6 +186,39 @@ virsh_find_all_interfaces_for_cgroup() {
...
+    local pid="${1}"
+    [ -z "${pid}" ] && return 1
+
+    local nsid=$(lsns -t net -p ${pid} -o NETNSID -nr)
+    [ -z "${nsid}" -o "${nsid}" = "unassigned" ] && return 1
+
+    set_source "netnsid"
+    ip link show |\
+        grep -B 1 -E " link-netnsid ${nsid}($| )" |\
+        ip link show | grep -B 1 -E " link-netnsid ${nsid}($| )" | sed -n -e "s|^[[:space:]]*[0-9]\+:[[:space:]]\+\([A-Za-z0-9_]\+\)\(@[A-Za-z0-9_]\+\)*:[[:space:]].*$|\1|p"
```
Comment (paulfantom) [inline collectors/cgroups.plugin/cgroup-network-helper.sh]: why is this piping `ip link` to `grep` to `ip link` to `grep`?

Comment (paulfantom) [inline collectors/cgroups.plugin/cgroup-network-helper.sh]: why is this piping `ip link` to `grep` to `ip link` to `grep`?

Comment (ktsaou) [reply collectors/cgroups.plugin/cgroup-network-helper.sh]: oops! just merged it!
This is wrong of course. Making a new PR...

Comment (ktsaou) [reply collectors/cgroups.plugin/cgroup-network-helper.sh]: https://github.com/netdata/netdata/pull/5417

**Comment by ShagoY** [comment]
I got the same error "cgroup-network-helper.sh: DEBUG: virsh command is not available" with the lastest docker image.

**Pull request merged**